Cargo.toml

```diff
@@ -66,6 +66,16 @@ transparent = { package = "zcash_transparent", version = "0.2" }
 zcash_keys = "0.8"
 zcash_primitives = "0.22"
 
+# Zcash chain state
+zaino-fetch = { git = "https://github.com/zingolabs/zaino.git", rev = "2b2ebb4725d771724c724587e2dfe44af7b46377" }
+zaino-proto = { git = "https://github.com/zingolabs/zaino.git", rev = "2b2ebb4725d771724c724587e2dfe44af7b46377" }
+zaino-state = { git = "https://github.com/zingolabs/zaino.git", rev = "2b2ebb4725d771724c724587e2dfe44af7b46377" }
+# - We do not use branch dependencies in general because they are brittle, but are forced
+#   to do so here because Zaino does.
+zebra-chain = { git = "https://github.com/ZcashFoundation/zebra.git", branch = "main" }
+zebra-rpc = { git = "https://github.com/ZcashFoundation/zebra.git", branch = "main" }
+zebra-state = { git = "https://github.com/ZcashFoundation/zebra.git", branch = "main" }
```

Comment:
We gain an awful lot of dependencies as a result of these, including a bunch of duplicate Zcash crates. It would be preferable to wait until at least either https://github.com/ZcashFoundation/zebra/pull/9300 or https://github.com/ZcashFoundation/zebra/pull/9369 is merged.

Reply:
I've addressed all the dependency reduction that I can achieve in the scope of this PR.